src/System.IO.FileSystem.AccessControl/tests/DirectoryObjectSecurityTests.cs

```diff
@@ -594,34 +571,35 @@ public void ResetAccessRule_AccessControlType_Deny_Succeeds()
             var customAccessRuleReadWrite = new CustomAccessRule(
                 identityReference, ReadWriteAccessMask, true, InheritanceFlags.None,
                 PropagationFlags.None, objectTypeGuid, Guid.NewGuid(), AccessControlType.Deny
                 );
 
             var customAccessRuleNetworkService = new CustomAccessRule(
-                new NTAccount(@"NT AUTHORITY\NetworkService"), SynchronizeAccessMask, true, InheritanceFlags.None,
+                new SecurityIdentifier(WellKnownSidType.NetworkServiceSid, null).Translate(typeof(NTAccount)), SynchronizeAccessMask, true, InheritanceFlags.None,
                 PropagationFlags.None, objectTypeGuid, Guid.NewGuid(), AccessControlType.Allow
                 );
 
             var customAccessRuleWrite = new CustomAccessRule(
-                new NTAccount(@"NT AUTHORITY\SYSTEM"), WriteAccessMask, true, InheritanceFlags.None,
+                new SecurityIdentifier(WellKnownSidType.LocalSystemSid, null).Translate(typeof(NTAccount)), WriteAccessMask, true, InheritanceFlags.None,
                 PropagationFlags.None, objectTypeGuid, Guid.NewGuid(), AccessControlType.Deny
                 );
 
             customObjectSecurity.AddAccessRule(customAccessRuleReadWrite);
-            customObjectSecurity.AddAccessRule(customAccessRuleNetworkService);
-            customObjectSecurity.ResetAccessRule(customAccessRuleWrite);
+            Assert.Contains(customAccessRuleReadWrite, customObjectSecurity.GetAccessRules(true, true, typeof(System.Security.Principal.NTAccount)).Cast<CustomAccessRule>());
 
-            AuthorizationRuleCollection ruleCollection =
-              customObjectSecurity
-             .GetAccessRules(true, true, typeof(System.Security.Principal.NTAccount));
+            customObjectSecurity.AddAccessRule(customAccessRuleNetworkService);
+            List<CustomAccessRule> existingRules = customObjectSecurity.GetAccessRules(true, true, typeof(System.Security.Principal.NTAccount)).Cast<CustomAccessRule>().ToList();
+            Assert.Contains(customAccessRuleReadWrite, existingRules);
+            Assert.Contains(customAccessRuleNetworkService, existingRules);
 
-            List<CustomAccessRule> existingRules = ruleCollection.Cast<CustomAccessRule>().ToList();
-            Assert.False(existingRules.Contains(customAccessRuleReadWrite));
-            Assert.False(existingRules.Contains(customAccessRuleNetworkService));
-            Assert.True(existingRules.Contains(customAccessRuleWrite));
+            customObjectSecurity.ResetAccessRule(customAccessRuleWrite);
+            existingRules = customObjectSecurity.GetAccessRules(true, true, typeof(System.Security.Principal.NTAccount)).Cast<CustomAccessRule>().ToList();
+            Assert.DoesNotContain(customAccessRuleReadWrite, existingRules);
+            Assert.Contains(customAccessRuleNetworkService, existingRules);
```

Comment:
Same here should it be DoesNotContain

Reply:
`Assert.False(existingRules.Contains(customAccessRuleNetworkService));` replaced by inverted `Assert.Contains(customAccessRuleNetworkService, existingRules);` because original fails test and I think the original test was wrong!